docs/content/3.cookbook/2.pass-custom-headers-and-cookies-to-requests.md

```diff
@@ -0,0 +1,266 @@
+```
+
+step 5.
+### **Initialize and Export the SDK**
+
+Finally, initialize the SDK with your configuration and export it.
+
+```javascript
+return initSDK<typeof sdkConfig>(sdkConfig);
+```
+
+step 6.
+### **sdk.config.ts** all together
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
+
+// Maintain a reference to the interceptor
+let interceptorId: number | null = null;
+
+export const useSdk = () => {
+  const sdkConfig = {
+    boilerplate: buildModule<BoilerplateModuleType>(boilerplateModule, {
+      apiUrl: 'http://localhost:8181/boilerplate',
+    }),
+  };
+
+  const {addHeadersToState, headerData} = useHeaders()
+
+  addHeadersToState()
+
+  // If an interceptor is already set, eject it to remove it
+  // this prevents headers being shared across instances
+  if (interceptorId !== null) {
+    client.interceptors.request.eject(interceptorId);
+  }
+
+  interceptorId = client.interceptors.request.use(
+    (config) => {
+      if (!config.headers) {
+        config.headers = {};
+      }
+
+      // here you can set any headers you want
+      config.headers['amazing-header'] = 'coolest header ever';
+      config.headers['dev-mode'] = headerData.value?.isDevMode ?? 'false'
+
+      return config;
+    },
+    (error) => {
+      // Do something with request error
+      return Promise.reject(error);
+    }
+  );
+
+  return initSDK<typeof sdkConfig>(sdkConfig);
+};
+
+```
+::tip
+This guide uses a `useHeaders` composable. This composable manages both server-side and client-side headers for you. This is just an example, you can use any method you want to fetch the headers.
+::
+
+step 7.
+**Create a Composable to Handle Headers**
+
+```javascript
+export default function () {
+  const serverHeaders = useRequestHeaders(['cookie']);
+  const devModeCookie = useCookie('dev-mode')?.value || '';
+
+  const state = useState<{ headerData: null | { isDevMode: string }, loading: boolean }>('headerState', () => ({
+    headerData: null,
+    loading: false,
+  }));
+
+  function addHeadersToState() {
+    getDevModeHeader(devModeCookie, serverHeaders.cookie || '');
+  }
+
+  function getHeaderObj(headers: string): { [key: string]: string } {
+    if (!headers) return {};
+
+    return headers.split(';').reduce<{ [key: string]: string }>((acc, item) => {
+      const parts = item.trim().split('=');
+      if (parts.length === 2) {
+        const [key, value] = parts;
+        acc[key] = value;
+      }
+      return acc;
+    }, {});
+  }
+
+  function getDevModeHeader(cookieHeader: string, serverHeader: string) {
+    if (process.server) {
+      const headerObj = getHeaderObj(serverHeader);
+      state.value.headerData = { isDevMode: headerObj['dev-mode'] || '' };
+      return
+    }
+
+    state.value.headerData = { isDevMode: cookieHeader };
+  }
+
+  return {
+    addHeadersToState,
+    ...toRefs(state.value),
+  }
+}
```

Comment:
Should we provide it ootb, instead of asking user to write it themselves. We even have the code already.

Reply:
I think that would be great. But I created the example quickly and I'm not certain there aren't edge cases that need to be addressed. This PR is a bit old, so Unified wasn't on my radar when I wrote the guide and we didn't have a storefront in which I could add it. But now we do, so I think it would indeed be great to have this ootb in Unified.

Reply:
@jagoral wdyt